week1/plot_trips.R

```diff
@@ -18,25 +24,47 @@ load('trips.RData')
 ########################################
 
 # plot the distribution of trip times across all rides (compare a histogram vs. a density plot)
-
+ggplot(data = trips, mapping = aes(x = tripduration)) + geom_histogram() + scale_x_log10(label = comma) + scale_y_log10(label = comma)
+ggplot(data = trips, mapping = aes(x = tripduration)) + geom_density() + scale_x_log10(label = comma) + scale_y_log10(label = comma)
+install.packages('patchwork')
+library(patchwork)
+plot1 <- ggplot(data = trips, mapping = aes(x = tripduration)) + geom_histogram() + scale_x_log10(label = comma) + scale_y_log10(label = comma)
+plot2 <- ggplot(data = trips, mapping = aes(x = tripduration)) + geom_density() + scale_x_log10(label = comma) + scale_y_log10(label = comma)
+plot1 + plot2
 # plot the distribution of trip times by rider type indicated using color and fill (compare a histogram vs. a density plot)
-
+ggplot(data = trips, mapping = aes(x = tripduration)) + geom_histogram(mapping = aes(color = usertype)) + scale_x_log10(label = comma) + scale_y_log10(label = comma)
+ggplot(data = trips, mapping = aes(x = tripduration)) + geom_density(mapping = aes(color = usertype)) + scale_x_log10(label = comma) + scale_y_log10(label = comma)
 # plot the total number of trips on each day in the dataset
-
+num_rides_per_day <- group_by(trips, ymd) %>% summarize(num_rides = n())
+ggplot(num_rides_per_day, aes(x = ymd, y = num_rides)) + geom_point()
 # plot the total number of trips (on the y axis) by age (on the x axis) and gender (indicated with color)
-
+age <- mutate(trips, age = 2022 - birth_year)
+ggplot(data = age, aes(x = age)) + geom_histogram(mapping = aes(color = gender))
 # plot the ratio of male to female trips (on the y axis) by age (on the x axis)
 # hint: use the pivot_wider() function to reshape things to make it easier to compute this ratio
 # (you can skip this and come back to it tomorrow if we haven't covered pivot_wider() yet)
+trips <- mutate(trips, age = 2022 - birth_year)
+trips_by_bd_gender <- trips %>% group_by(age, gender) %>% summarize(count = n())
+trips_by_bd_gender <- pivot_wider(trips_by_bd_gender, names_from = gender, values_from = count)
+trips_by_bd_gender <- select(trips_by_bd_gender, age, Male, Female)
+trips_by_bd_gender <- mutate(trips_by_bd_gender, ratio = Male / Female)
+ggplot(data = trips_by_bd_gender, mapping = aes(x = age, y = ratio)) + geom_point()
```

Comment:
You kinda fell into an object oriented way of doing things here. The pipe syntax would've made this a lot cleaner.